public/js/twiliochat.js

```diff
@@ -90,12 +90,15 @@ var twiliochat = (function() {
   function connectMessagingClient(tokenResponse) {
```

Comment:
This method is called from `fetchAccessToken`, right?

What about refactoring `fetchAccessToken` into something like this:

```javascript
function fetchAccessToken(username, handler) {
  $.post("/token", { identity: username, device: "browser" }, null, "json")
    .done(function(resp) {
      handler(resp.token); // We pass only the "token" to "connectMessagingClient".
    });
}
```

`connectMessagingClient` doesn't need to know about the inner structure of the object that contains the token which is the benefit of doing this refactor. I'd like to hear your thoughts on this.